File: Examples/ClassDiagram/TypeScript/Playable.ts
/** Anything the player knows how to play. */
export interface Playable {
    readonly title: string;
    readonly duration: number;

    play(): void;
}

/** A coarse classification for a piece of media. */
export enum Genre {
    Pop,
    Rock,
    Jazz,
    Classical,
    Electronic,
    SpokenWord,
}

File: Examples/ClassDiagram/Swift/Playable.swift
import Foundation

/// Anything the player knows how to play.
public protocol Playable {
    var title: String { get }
    var duration: TimeInterval { get }

    func play()
}

/// A coarse classification for a piece of media.
public enum Genre {
    case pop
    case rock
    case jazz
    case classical
    case electronic
    case spokenWord
}
